engine/Dockerfile

```diff
@@ -27,7 +27,7 @@ RUN if [ "$TARGETPLATFORM" = "linux/arm64" ]; then \
     && rm grpcio-1.64.1-cp312-cp312-linux_aarch64.whl; \
     fi
 
-RUN pip install uv
+RUN pip install uv setuptools
```

Comment:
if setuptools is being added through requirements, does it need to be here too? (same above)

Reply:
I guess _technically_ no, but I think `setuptools` is only being included in `requirements.txt`/`requirements-dev.txt` because of two transitive dependencies. If those dependencies drop their requirement on `setuptools` we'll need this anyways 🤔 

Python 3.12's release notes [mention](https://docs.python.org/3/whatsnew/3.12.html#:~:text=The%20third%2Dparty%20Setuptools%20package%20continues%20to%20provide%20distutils%2C%20if%20you%20still%20require%20it%20in%20Python%203.12%20and%20beyond):
> https://github.com/python/cpython/issues/95299: Do not pre-install setuptools in virtual environments created with [venv](https://docs.python.org/3/library/venv.html#module-venv). This means that distutils, setuptools, pkg_resources, and easy_install will no longer available by default; to access these run pip install setuptools in the [activated](https://docs.python.org/3/library/venv.html#venv-explanation) virtual environment.